features/mbedtls/platform/COMPONENT_PSA_SRV_IMPL/src/default_random_seed.cpp

```diff
@@ -5,7 +5,16 @@
 
 int mbed_default_seed_read(unsigned char *buf, size_t buf_len)
 {
-    psa_its_status_t rc = psa_its_get(MBED_RANDOM_SEED_ITS_UID, 0, buf_len, buf);
+    psa_its_status_t rc;
+    struct psa_its_info_t info = {0, 0};
+    size_t actual_size = buf_len;
+    psa_its_get_info(MBED_RANDOM_SEED_ITS_UID, &info);
+    if (info.size < buf_len)
+    {
+        actual_size = info.size;
```

Comment:
Please note that `mbed_default_seed_read(unsigned char *buf, size_t buf_len)` is not able to tell anyone it has read less then required as `buf_len` is an **IN** argument only.
Thus reading less then requested is not a solution.
Consider calling panic as it looks like missconfiguration issue. System integrator MUST know what seed size is needed for his system, and must inject appropriate size.
By gracefully treating this error we may introduce security vulnerabilities, when error code will not be treated properly.